Gem/Settings/ViewModels/ChainSettingsViewModel.swift

```diff
@@ -4,54 +4,149 @@ import Foundation
 import Primitives
 import GemstonePrimitives
 import Settings
+import Blockchain
+import BigInt
 
-class ChainSettingsViewModel: ObservableObject {
-    
-    let chain: Chain
+@Observable
+class ChainSettingsViewModel {
+    private typealias ChainFetchData = (chainNode: ChainNode, latency: LatencyMeasureService.Latency?, blockNumber: BigInt?, error: Error?)
+
+    let explorerService: ExplorerService
     let nodeService: NodeService
-    
-    @Published var chainNode: ChainNode {
-        didSet {
-            try? nodeService.setNodeSelected(chain: chain, node: chainNode.node)
-        }
-    }
-    @Published var nodes: [ChainNode] = []
-    
-    @Published var selectedExplorer: String? = .none
+
+    let chain: Chain
+
+    var isPresentingImportNode: Bool = false
+
+    var selectedExplorer: String?
     var explorers: [String]
-    let explorerService: ExplorerService
-    
+
+    var selectedNode: ChainNode
+    var nodeDelete: ChainNode?
+
+    private var nodes: [ChainNode] = []
+    private var nodesChainIdData: [String: ChainFetchData] = [:]
+
     init(
-        chain: Chain,
         nodeService: NodeService,
-        explorerService: ExplorerService
+        explorerService: ExplorerService,
+        chain: Chain
     ) {
-        self.chain = chain
         self.nodeService = nodeService
         self.explorerService = explorerService
-        self.chainNode = nodeService.getNodeSelected(chain: chain)
+
+        self.chain = chain
+        self.selectedNode = nodeService.getNodeSelected(chain: chain)
+
         self.explorers = ExplorerService.explorers(chain: chain)
         self.selectedExplorer = explorerService.get(chain: chain) ?? explorers.first
     }
-    
-    var title: String {
-        Asset(chain).name
+
+    var title: String { Asset(chain).name }
+
+    var nodesTitle: String { Localized.Settings.Networks.source }
+    var nodesModels: [ChainNodeViewModel] {
+        nodes.map { node in
+            let chainData = nodesChainIdData[node.id]
+            return ChainNodeViewModel(
+                chainNode: node,
+                blockNumber: chainData?.blockNumber ?? .none,
+                latency: chainData?.latency ?? .none,
+                blockNumberError: chainData?.error
+            )
+        }
     }
 
+    var explorerTitle: String { Localized.Settings.Networks.explorer }
+
     var isSupportedAddingCustomNode: Bool {
         AssetConfiguration.addCustomNodeChains.contains(chain.type)
     }
+
+    func canDelete(chainNode: ChainNode) -> Bool {
+        guard nodes.count > 1 else { return false }
+        return !chainNode.isGemNode
+    }
 }
 
 // MARK: - Business logic
 
 extension ChainSettingsViewModel {
-    func getNodes() throws -> [ChainNode] {
-        try nodeService.nodes(for: chain)
+    func fetch() async throws {
+        try await MainActor.run { [self] in
```

Comment:
can we avoid async here? sqlite calls are instant no need to wrap into async, too much complexity